jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/annotations/ErrorAnnotator.kt

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package software.aws.toolkits.jetbrains.services.cloudformation.annotations
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.ExternalAnnotator
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiFile
+
+/**
+ *
+ * Process the files in a language by an external annotation tool ("linter").
+ * External annotators are expected to be (relatively) slow and are started
+ * after regular annotators have completed their work.
+ *
+ * Annotators work in three steps:
+ * 1. [.collectInformation] is called to collect some data about a file needed for launching a tool
+ * 2. collected data is passed to [.doAnnotate] which executes a tool and collect highlighting data
+ * 3. highlighting data is applied to a file by [.apply]
+ *
+ */
+class ErrorAnnotator : ExternalAnnotator<InitialAnnotationResults, List<ErrorAnnotation>>() {
```

Comment:
ErrorAnnotator is a very generic name, since it's loaded into the IDE as an extension can we give it a full name like `CloudFormationLintAnnotator`